src/lib/helpers.ts

```diff
@@ -344,3 +356,551 @@ export function parseResponseAPIVersion(response: Response) {
+    },
+    excludeCredentials: optionsJSON.excludeCredentials?.map(toPublicKeyCredentialDescriptor),
+  }
+
+  // Finalize options
+  const options: CredentialCreationOptions = { publicKey }
+  // Set up the ability to cancel this request if the user attempts another
+  options.signal = WebAuthnAbortService.createNewAbortSignal()
+
+  // Wait for the user to complete attestation
+  let credential
+  try {
+    credential = (await navigator.credentials.create(options)) as RegistrationCredential
+  } catch (err) {
+    throw identifyRegistrationError({ error: err as Error, options })
+  }
+
+  if (!credential) {
+    throw new Error('Registration was not completed')
+  }
+
+  const { id, rawId, response, type } = credential
+
+  // Continue to play it safe with `getTransports()` for now, even when L3 types say it's required
+  let transports: AuthenticatorTransportFuture[] | undefined = undefined
+  if (typeof response.getTransports === 'function') {
+    transports = response.getTransports()
+  }
+
+  // L3 says this is required, but browser and webview support are still not guaranteed.
+  let responsePublicKeyAlgorithm: number | undefined = undefined
+  if (typeof response.getPublicKeyAlgorithm === 'function') {
+    try {
+      responsePublicKeyAlgorithm = response.getPublicKeyAlgorithm()
+    } catch (error) {
+      warnOnBrokenImplementation('getPublicKeyAlgorithm()', error as Error)
+    }
+  }
+
+  let responsePublicKey: string | undefined = undefined
+  if (typeof response.getPublicKey === 'function') {
+    try {
+      const _publicKey = response.getPublicKey()
+      if (_publicKey !== null) {
+        responsePublicKey = bufferToBase64URLString(_publicKey)
+      }
+    } catch (error) {
+      warnOnBrokenImplementation('getPublicKey()', error as Error)
+    }
+  }
+
+  // L3 says this is required, but browser and webview support are still not guaranteed.
+  let responseAuthenticatorData: string | undefined
+  if (typeof response.getAuthenticatorData === 'function') {
+    try {
+      responseAuthenticatorData = bufferToBase64URLString(response.getAuthenticatorData())
+    } catch (error) {
+      warnOnBrokenImplementation('getAuthenticatorData()', error as Error)
+    }
+  }
+
+  return {
+    id,
+    rawId: bufferToBase64URLString(rawId),
+    response: {
+      attestationObject: bufferToBase64URLString(response.attestationObject),
+      clientDataJSON: bufferToBase64URLString(response.clientDataJSON),
+      transports,
+      publicKeyAlgorithm: responsePublicKeyAlgorithm,
+      publicKey: responsePublicKey,
+      authenticatorData: responseAuthenticatorData,
+    },
+    type,
+    clientExtensionResults: credential.getClientExtensionResults(),
+    // authenticatorAttachment: toAuthenticatorAttachment(credential.authenticatorAttachment),
+  }
+}
+
+/**
+ * Attempt to intuit _why_ an error was raised after calling `navigator.credentials.create()`
+ */
+export function identifyRegistrationError({
```

Comment:
Are these lifted directly from simplewebauthn? I'd prefer if we didn't do any additional error handling other than what the browser returns. Most of these are not relevant for the MFA use case anyway.

Reply:
Yes directly lifted - we can remove yup